deps/openssl/openssl/ssl/ssl_locl.h

```diff
@@ -1125,9 +1125,12 @@ struct ssl_st {
     void (*msg_callback) (int write_p, int version, int content_type,
                           const void *buf, size_t len, SSL *ssl, void *arg);
     void *msg_callback_arg;
-    int (*key_callback)(SSL *ssl, int name, const unsigned char *secret,
-                        size_t secretlen, void *arg);
-    void *key_callback_arg;
+    int (*encryption_secrets_callback)(SSL* ssl, int level,
+                                       const uint8_t *read_secret,
+                                       const uint8_t *write_secret,
+                                       size_t secret_len,
+                                       void *arg);
+    void *encryption_secrets_callback_arg;
```

Comment:
btw, I *think* these changes are actually ABI breaking in openssl.

Reply:
The changes to `deps/openssl/openssl/include` are ABI- and API-breaking, yes … if you want compatibility, I think you would ideally want to retain `key_callback` (and its setter), and fall back to using it when no `encryption_secrets_callback` is set, if that’s possible

Reply:
key_callback is not part of the OpenSSL API. tatsuhiro-san added those in his initial hack to get QUIC support working. This is replacing his non-standard api with another non-standard api

Reply:
I think we’re good then – C function additions generally are ABI- and API-stable, and anything that doesn’t touch the include folder is okay, too.